python/itools-y4m.py

```diff
@@ -14,34 +14,27 @@
 itools_common = importlib.import_module("itools-common")
 
 
-def color_range_conversion(inyvu, input_colorrange, output_colorrange):
+def color_range_conversion(inyvu, input_colorrange, output_colorrange, colorspace):
     # get components
     ya = inyvu[:, :, 0]
     va = inyvu[:, :, 1]
     ua = inyvu[:, :, 2]
     ya, ua, va, status = color_range_conversion_components(
-        ya, ua, va, input_colorrange, output_colorrange
+        ya, ua, va, input_colorrange, output_colorrange, colorspace,
     )
     outyvu = np.stack((ya, va, ua), axis=2)
     return outyvu
 
 
-def color_range_conversion_components(ya, ua, va, input_colorrange, output_colorrange):
-    ya, ya_broken = luma_range_conversion(
-        ya,
-        src=input_colorrange,
-        dst=output_colorrange,
-    )
-    ua, ua_broken = chroma_range_conversion(
-        ua,
-        src=input_colorrange,
-        dst=output_colorrange,
-    )
-    va, va_broken = chroma_range_conversion(
-        va,
-        src=input_colorrange,
-        dst=output_colorrange,
-    )
+def color_range_conversion_components(ya, ua, va, input_colorrange, output_colorrange, colorspace):
+
+    srcmin, srcmax, _ = get_colorspace_min_max(colorspace, input_colorrange.name)
+    dstmin, dstmax, dt = get_colorspace_min_max(colorspace, output_colorrange.name)
+    # luma_range_conversion
+    ya, ya_broken = do_range_conversion(ya, srcmin, srcmax, dstmin, dstmax, dt)
+    # chroma_range_conversion
+    ua, ua_broken = do_range_conversion(ua, srcmin, srcmax, dstmin, dstmax, dt)
```

Comment:
This is still broken. The range of values for limited range is different for the luma and the chromas

Reply:
I  fixed it in the latest commit, this commit is outdated.